client/web/src/enterprise/batches/backend.ts

```diff
@@ -212,10 +214,23 @@ export const BATCH_CHANGES_SITE_CONFIGURATION = gql`
     }
 `
 
-export const useGetBatchChangesSiteConfiguration = (): QueryResult<
-    BatchChangesSiteConfigurationResult,
-    BatchChangesSiteConfigurationVariables
-> =>
-    useQuery(BATCH_CHANGES_SITE_CONFIGURATION, {
+interface useGetBatchChangesSiteConfigurationResult {
+    loading: boolean
+    error: ApolloError | undefined
+    rolloutWindowConfig: BatchChangeRolloutWindow[]
+}
+
+export const useBatchChangesRolloutWindowConfig = (): useGetBatchChangesSiteConfigurationResult => {
+    const [rolloutWindowConfig, setRolloutWindowConfig] = useState<BatchChangeRolloutWindow[]>([])
+    const { loading, error, data } = useQuery(BATCH_CHANGES_SITE_CONFIGURATION, {
         fetchPolicy: 'cache-first',
     })
+    useEffect(() => {
+        if (data) {
+            const siteConfig = jsonc.parse(data.site.configuration.effectiveContents)
+            setRolloutWindowConfig(siteConfig['batchChanges.rolloutWindows'] || [])
+        }
+    }, [data])
+
+    return { loading, error, rolloutWindowConfig }
```

Comment:
Keeping the config in state and running the `useEffect` to set it means unnecessary re-renders here. What if we try to return memoize parsing it from the JSON and then return it directly:

```suggestion
    const { loading, error, data } = useQuery(BATCH_CHANGES_SITE_CONFIGURATION, {
        fetchPolicy: 'cache-first',
    })
   
   const rolloutWindowConfig: BatchChangeRolloutWindow[] = useMemo(() => {
        if (!data) {
            return []
        }
        const siteConfig = jsonc.parse(data.site.configuration.effectiveContents)
        return siteConfig['batchChanges.rolloutWindows'] || []
    }, [data])

    return { loading, error, rolloutWindowConfig }
```

Reply:
Oh! This is much better.

TIL